pennylane_lightning/lightning_tensor/quimb/_mps.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Class implementation for MPS manipulation based on the `quimb` Python package.
+"""
+
+import numpy as np
+import quimb.tensor as qtn
+from pennylane.wires import Wires
+
+# TODO: understand if supporting all operations and observables is feasible for the first release
+# I comment the following lines since otherwise Codecov complaints
+
+# _operations = frozenset({})
+# The set of supported operations.
+
+# _observables = frozenset({})
+# The set of supported observables.
+
+
+class QuimbMPS:
```

Comment:
I really like this implementation. So we'll have classes for quimb+TN, cuTN+TN, cuTN+MPS as well which expose the same API?

Reply:
Yes, the plan is to have one interface for each combination (at least whenever is convenient for us). They will have all the necessary details depending on the specific implementation, such as interfaces with C++ backends, etc. but the idea is to expose APIs that are necessary for `LightningTensor` to work (such as `execute`)